### ISSUE TITLE
Search: Accessibility issues with screen reader

### ISSUE THREAD

**Issue by ryelle** [WordPress/wporg-mu-plugins#118]
> Missing labels and no way to tell if Search link did anything. Not meeting standards.

https://wordpress.slack.com/archives/C02QB8GMM/p1642993201177300?thread_ts=1642992375.174000&cid=C02QB8GMM

The way we're hacking the nav menu/submenu behavior works visually, but is confusing for a screen reader user - there's a search link, but it goes nowhere. The input itself has a label attached correctly (id/for), but it's hidden using `display: none` so it's ignored.
